comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +24,97 @@ public static bool IsDirectory(string path)
+        }
 
-		// Return the nesting depth of the given directory. A directory containing only files (no subdirectories) has a depth of 0.
-		public static int GetDepth(string directory)
+        // Return the nesting depth of the given directory. A directory containing only files (no subdirectories) has a depth of 0.
+        public static int GetDepth(string directory)
 		{
-			throw new NotImplementedException();
-		}
+            //Return 0 if directory is null
+            if (string.IsNullOrEmpty(directory))
+                return 0;
+            //Return 1 if there's no parent
+            DirectoryInfo parent = Directory.GetParent(directory);
+            if (parent == null)
+                return 1;
+            
+            return GetDepth(parent.FullName) + 1;
+        }
 
 		// Get the path and size (in bytes) of the smallest file below the given directory
 		public static Tuple<string, long> GetSmallestFile(string directory)
 		{
-			throw new NotImplementedException();
-		}
+            string[] files = Directory.GetFiles(directory, "*.*", SearchOption.AllDirectories);
+            List<Tuple<string, long>> fileSizes = new List<Tuple<string, long>>();
+            //Add the name of the files and their size to the list
+            foreach (string name in files)
+            {
+                FileInfo info = new FileInfo(name);
+                fileSizes.Add(new Tuple<string, long>(info.FullName, info.Length));
+            }
+            //Sort the list by the sizes (Item2 of the tuple)
+            fileSizes.Sort(
+                delegate (Tuple<string,long> a,Tuple<string,long> b)
+                {
+                    return a.Item2.CompareTo(b.Item2);
+                }
+            );
+            return fileSizes[0]; //Return the first index since it will be the largest after sorting.
+        }
 
-		// Get the path and size (in bytes) of the largest file below the given directory
-		public static Tuple<string, long> GetLargestFile(string directory)
+        // Get the path and size (in bytes) of the largest file below the given directory
+        public static Tuple<string, long> GetLargestFile(string directory)
 		{
-			throw new NotImplementedException();
-		}
+            string[] files = Directory.GetFiles(directory, "*.*", SearchOption.AllDirectories);
+            List<Tuple<string, long>> fileSizes = new List<Tuple<string, long>>();
+            //Add the name of the files and their size to the list
+            foreach (string name in files)
+            {
+                FileInfo info = new FileInfo(name);
+                fileSizes.Add(new Tuple<string, long>(info.FullName, info.Length));
+            }
+            //Sort the list by the sizes (Item2 of the tuple)
+            fileSizes.Sort(
+                delegate (Tuple<string, long> a, Tuple<string, long> b)
+                {
+                    return b.Item2.CompareTo(a.Item2);
```

Comment:
Worth adding a comment here that you're switching `a` and `b` to get the list sorted in descending order